packages/lwc/types.d.ts

```diff
@@ -193,17 +193,17 @@ declare module 'lwc' {
      * @param adapter the adapter used to provision data
      * @param config configuration object for the adapter
      */
-    export function wire(
-        adapter: WireAdapterConstructor | LegacyWireAdapterConstructor,
-        config?: WireConfigValue
+    export function wire<T, R>(
+        adapter: WireAdapterConstructor<T> | LegacyWireAdapterConstructor<T, R>,
+        config?: T
```

Comment:
My TypeScript-Fu is weak, but if I'm not mistaken, isn't this broadening rather than narrowing the allowed type here? Should it be something like `config?: T extends WireConfigValue` instead?

Reply:
Adding another comment of where I think that should be applied. (WireAdapter definition)